src/main/java/org/takes/facets/auth/social/LinkedinProfileJson.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.takes.facets.auth.social;
+
+import com.jcabi.http.request.JdkRequest;
+import com.jcabi.http.response.JsonResponse;
+import com.jcabi.http.response.RestResponse;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.util.Iterator;
+import javax.json.JsonObject;
+import org.takes.misc.Href;
+
+/**
+ * LinkedIn member profile in json format.
+ *
+ * <p>The class is immutable and thread-safe.
+ * @author Igor Khvostenkov (ikhvostenkov@gmail.com)
+ * @version $Id$
+ * @since 0.14.4
+ * @checkstyle MultipleStringLiteralsCheck (500 lines)
+ */
+public final class LinkedinProfileJson implements MemberProfileJson {
+    /**
+     * App name.
+     */
+    private final transient String app;
+
+    /**
+     * Key.
+     */
+    private final transient String key;
+
+    /**
+     * Ctor.
+     * @param sapp Social network app
+     * @param skey Social network key
+     */
+    public LinkedinProfileJson(final String sapp, final String skey) {
+        this.app = sapp;
+        this.key = skey;
+    }
+
+    @Override
+    public JsonObject fetch(final String tokenurl, final String socialurl,
+                            final Href home) throws IOException {
```

Comment:
@yegor256 IDE default carriage return. Align it with the beginning of the next line?